pkg/graveler/graveler_v2_test.go

```diff
@@ -111,6 +119,83 @@ func TestGravelerGet(t *testing.T) {
+
+		val, err := test.Sut.Get(ctx, repository, graveler.Ref(branch3ID), key1)
+
+		require.NoError(t, err)
+		require.NotNil(t, val)
+		require.Equal(t, value1, val)
+	})
+
+	t.Run("get from branch - compacted with staged only flag when get object committed", func(t *testing.T) {
+		test := testutil.InitGravelerTest(t)
+		test.RefManager.EXPECT().ParseRef(graveler.Ref(branch3ID)).Times(1).Return(rawRefBranch3, nil)
+		test.RefManager.EXPECT().ResolveRawRef(ctx, repository, rawRefBranch3).Times(1).Return(&graveler.ResolvedRef{Type: graveler.ReferenceTypeBranch, BranchRecord: graveler.BranchRecord{BranchID: branch3ID, Branch: &branch3}}, nil)
+
+		test.StagingManager.EXPECT().Get(ctx, stagingToken1, key1).Times(1).Return(nil, graveler.ErrNotFound)
+		test.StagingManager.EXPECT().Get(ctx, stagingToken2, key1).Times(1).Return(nil, graveler.ErrNotFound)
+		test.StagingManager.EXPECT().Get(ctx, stagingToken3, key1).Times(1).Return(nil, graveler.ErrNotFound)
+
+		test.CommittedManager.EXPECT().Get(ctx, repository.StorageNamespace, mr2ID, key1).Times(1).Return(value1, nil)
+		test.RefManager.EXPECT().GetCommit(ctx, repository, commit1ID).Times(1).Return(&commit1, nil)
+
+		test.CommittedManager.EXPECT().Get(ctx, repository.StorageNamespace, commit1.MetaRangeID, key1).Times(1).Return(value1, nil)
+		val, err := test.Sut.Get(ctx, repository, graveler.Ref(branch3ID), key1, graveler.WithStageOnly(true))
+
+		require.Error(t, graveler.ErrNotFound, err)
+		require.Nil(t, val)
+	})
+
+	t.Run("get from branch - compacted with staged only flag when get object different in commit", func(t *testing.T) {
+		test := testutil.InitGravelerTest(t)
+		test.RefManager.EXPECT().ParseRef(graveler.Ref(branch3ID)).Times(1).Return(rawRefBranch3, nil)
+		test.RefManager.EXPECT().ResolveRawRef(ctx, repository, rawRefBranch3).Times(1).Return(&graveler.ResolvedRef{Type: graveler.ReferenceTypeBranch, BranchRecord: graveler.BranchRecord{BranchID: branch3ID, Branch: &branch3}}, nil)
+
+		test.StagingManager.EXPECT().Get(ctx, stagingToken1, key1).Times(1).Return(nil, graveler.ErrNotFound)
+		test.StagingManager.EXPECT().Get(ctx, stagingToken2, key1).Times(1).Return(nil, graveler.ErrNotFound)
+		test.StagingManager.EXPECT().Get(ctx, stagingToken3, key1).Times(1).Return(nil, graveler.ErrNotFound)
+
+		test.CommittedManager.EXPECT().Get(ctx, repository.StorageNamespace, mr2ID, key1).Times(1).Return(value1, nil)
+		test.RefManager.EXPECT().GetCommit(ctx, repository, commit1ID).Times(1).Return(&commit1, nil)
+
+		test.CommittedManager.EXPECT().Get(ctx, repository.StorageNamespace, commit1.MetaRangeID, key1).Times(1).Return(value2, nil)
+		val, err := test.Sut.Get(ctx, repository, graveler.Ref(branch3ID), key1, graveler.WithStageOnly(true))
+
+		require.NoError(t, err)
+		require.NotNil(t, val)
+		require.Equal(t, value1, val)
+	})
+
+	t.Run("get from branch - compacted with staged only flag when get object different in commit", func(t *testing.T) {
```

Comment:
What's the difference from the above?

Reply:
wrong test name
It's a scenario when the object isn't committed